supervisor/medusa/http_server.py

```diff
@@ -506,10 +506,13 @@ def maintenance (self):
 
     def kill_zombies (self):
         now = int (time.time())
-        for channel in list(asyncore.socket_map.values()):
-            if channel.__class__ == self.__class__:
-                if (now - channel.last_used) > channel.zombie_timeout:
-                    channel.close()
+        channels = list(asyncore.socket_map.values())
+        chan_len = len(channels)
```

Comment:
> Fix dictionary changed size during iteration

This code does not have a dictionary.  I also don't see why this change would make a difference, since `list(asyncore.socket_map.values())` should not change.  Can you please explain?

Reply:
oh, yes, you are right , my mistakes, it is the same. sorry about that.

since i met a bug :
```python
ERRO 0.0.0.0:uncaptured python exception, closing channel <supervisor.http.supervisor_af_inet_http_server listening 0.0.0.0:9001 at 0xb5fc5410> (<class 'RuntimeError'>:dictionary changed size during iteration 
...
http_server.py|kill_zombies|509])
```
Then i check the source code and ... . maybe this is another bug.
ps, my version is 4.0.0

THANKS FOR YOUR ANSWER

Reply:
> ps, my version is 4.0.0

In Supervisor 4.0.0, the line is [the same](https://github.com/Supervisor/supervisor/blob/4.0.0/supervisor/medusa/http_server.py#L509).  It does not iterate a dictionary, so I don't see how it could produce the error `dictionary changed size during iteration`.

Perhaps you have a development version like `4.0.0.dev0` from before we merged [this fix](https://github.com/Supervisor/supervisor/pull/1187/files)(?).  Can you please check line 509 in `http_server.py` of your local copy?

Reply:
YES, this is a error for my version
![image](https://user-images.githubusercontent.com/31585302/83723895-f333e080-a671-11ea-9f24-42eb9000a8c3.png)

Ok, then this bug has solved in new version

Thanks a lot.

Reply:
Great, thanks for following up!  I'll close this PR.